worlds/pokemon_emerald/__init__.py

```diff
@@ -0,0 +1,841 @@
+                fill_item_candidates_by_category[category].sort()
+
+            # Ignore vanilla occurrences and pick completely randomly
+            if self.multiworld.item_pool_type[self.player].value == ItemPoolType.option_diverse:
+                item_category_weights = [len(category_list) for category_list in fill_item_candidates_by_category.values()]
+
+            # TMs should not have duplicates until every TM has been used already
+            all_tm_choices = fill_item_candidates_by_category["TM"].copy()
+
+            def refresh_tm_choices() -> None:
+                fill_item_candidates_by_category["TM"] = all_tm_choices.copy()
+                self.random.shuffle(fill_item_candidates_by_category["TM"])
+
+            # Create items
+            for item in default_itempool:
+                if item.classification != ItemClassification.progression and "Unique" not in item.tags:
+                    category = self.random.choices(item_categories, item_category_weights)[0]
+                    if category == "TM":
+                        if len(fill_item_candidates_by_category["TM"]) == 0:
+                            refresh_tm_choices()
+                        item_code = fill_item_candidates_by_category["TM"].pop()
+                    else:
+                        item_code = self.random.choice(fill_item_candidates_by_category[category])
+                    item = self.create_item_by_code(item_code)
+
+                self.multiworld.itempool.append(item)
+
+    def set_rules(self) -> None:
+        set_default_rules(self.multiworld, self.player)
+
+        if self.multiworld.overworld_items[self.player].value == Toggle.option_true:
+            set_overworld_item_rules(self.multiworld, self.player)
+
+        if self.multiworld.hidden_items[self.player].value == Toggle.option_true:
+            set_hidden_item_rules(self.multiworld, self.player)
+
+        if self.multiworld.npc_gifts[self.player].value == Toggle.option_true:
+            set_npc_gift_rules(self.multiworld, self.player)
+
+        if self.multiworld.enable_ferry[self.player].value == Toggle.option_true:
+            set_enable_ferry_rules(self.multiworld, self.player)
+
+        if self.multiworld.require_itemfinder[self.player].value == Toggle.option_true:
+            add_hidden_item_itemfinder_rules(self.multiworld, self.player)
+
+        if self.multiworld.require_flash[self.player].value == Toggle.option_true:
+            add_flash_rules(self.multiworld, self.player)
+
+    def generate_basic(self) -> None:
+        victory_event_name = "EVENT_DEFEAT_CHAMPION"
+        if self.multiworld.goal[self.player] == Goal.option_steven:
+            victory_event_name = "EVENT_DEFEAT_STEVEN"
+        if self.multiworld.goal[self.player] == Goal.option_norman:
+            victory_event_name = "EVENT_DEFEAT_NORMAN"
+
+        self.multiworld.completion_condition[self.player] = lambda state: state.has(victory_event_name, self.player)
```

Comment:
this should really be in `set_rules`

Reply:
Fixed in 0299e0b. Was probably still here from referencing the example world where the event itself was created and placed here.